### DISCUSSION TITLE
How can I get a un-comitted build of the project?

### DISCUSSION THREAD

**Discussion by EvanCarroll** [semantic-release/semantic-release#2160] [Q&A] [answered]
Previously I was going down the process of how we develop software, (1) fork, (2) commit, (3) open PR, (4) QA (5) PR is commited, (6) final build is produced. It makes sense to me to have during the QA process a build created.

How is this supposed to work in `semantic-release`? It seems like `semantic-release` does nothing to help get a build out of a fork. So how does the alpha and beta channel work? Is there any documentation or examples on how were supposed to set up this flow such that you can get a build from the project before the work is commited? Do all developers commit to alpha and beta channels? How do these channels get sycn'd to the main branch? It seems like this is a global development branch that everyone shares. How does it get QA'd and what happens when commits are rejected by QA after they've been commited to Alpha?

And what happens to the changelog? It's created for the alpha-buiild and then you open a PR from the alpha to master and it totally gets rewritten? Is there any documentation about this?

**Comment by travi** [comment, marked as answer]
> How is this supposed to work in `semantic-release`?

it seems that you have some expectations of semantic-release that are outside the reality of the project. you have opened several threads about how semantic-release is not meeting your needs and commented on others with similar sentiment. i want to make it clear that semantic-release is not a general workflow tool. it does have quite a bit of flexibility and the details that it is able to automate can make it attractive for use as a more general purpose workflow tool than is it designed for.

as [mentioned in the thread about forking](https://github.com/semantic-release/semantic-release/discussions/2158#discussioncomment-1405282), semantic-release is not intended to support workflows that involve forking. the obvious exception to that statement is for accepting outside contributions, but semantic-release does not become involved in that process until a pull request is accepted into the original process and the changes are merged into a release branch.

release branches for a project managed by semantic-release are simply git branches within the primary project. the release branch for the stable/latest branch is long-lived. other release branches (pre-releases) are temporary and only live until promoted to the stable/latest branch. details of these options can be [found in our documentation](https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration)



> It makes sense to me to have during the QA process a build created.

we do not recommend a workflow that requires manual QA review for every release. this is even more true if the QA process you refer to means testing the changes of your package in a consuming project. instead, we recommend following development practices that give your team the confidence to release your package independent of any consumers. as [mentioned in our docs](https://semantic-release.gitbook.io/semantic-release/support/faq#is-it-really-a-good-idea-to-release-on-every-push) we encourage use of GitHub Flow and stress the importance of the rule about the mainline _always_ being deployable.



if the workflow you are targeting is not in line with the goals of semantic-release, it is possible that semantic-release might not actually be the tool you are looking for
